Gems/Atom/RHI/Code/Include/Atom/RHI/ObjectCollector.h

```diff
@@ -172,6 +179,38 @@ namespace AZ
             {
                 m_pendingGarbage.push_back({ AZStd::move(m_pendingObjects), m_currentIteration });
             }
+
+            if (m_pendingNotifies.size())
+            {
+                if (m_pendingGarbage.size())
+                {
+                    // find the newest garbage entry and add any pending notifies
+                    Garbage& latestGarbage = m_pendingGarbage.front();
+                    size_t latestGarbageAge = m_currentIteration - latestGarbage.m_collectIteration;
+                    size_t i = 1;
+                    while (i < m_pendingGarbage.size())
+                    {
```

Comment:
Am I crazy or should i be incremented in this loop? Not quite sure how this loop exits.

Reply:
It is incremented in the else clause though?
It will loop and remove garbage entries ( swap & pop ) and skip non ready ones

Reply:
Or at least it was that way in the original code in `ObjectCollector<Traits>::Collect`

Reply:
Yep this is missing a ++i;

Reply:
This is fixed, I originally pushed this as a discussion branch with @moudgils, but it has now been cleaned up and tested.